File: Design_Pattern_and_Principles/StrategyPatternExample/StrategyTest.java
public class StrategyTest {
    public static void main(String[] args) {
        PaymentContext context = new PaymentContext();
        context.setPaymentStrategy(new CreditCardPayment("1234567890123456"));
        context.checkout(120.75);
        context.setPaymentStrategy(new PayPalPayment("developer@example.com"));
        context.checkout(45.00);
    }
}

File: Week_1/Design_Pattern_and_Principles/StrategyPatternExample/StrategyTest.cs
public class StrategyTest {
    public static void Main(string[] args) {
        PaymentContext context = new PaymentContext();

        context.SetPaymentStrategy(new CreditCardPayment("1234567890123456"));
        context.Checkout(120.75);

        context.SetPaymentStrategy(new PayPalPayment("developer@example.com"));
        context.Checkout(45.00);
    }
}